### PR TITLE
Apply eslint max-len to strings and fix (most) warnings

### PR DESCRIPTION
**Pull request by reosarevok**
I skipped some files where the indentation is a mess and needs to be fixed before even knowing how long a line actually should be, and I don't doubt I might have missed a couple other files in the process, but this is big enough as it is.

### CODE REVIEW

File: root/account/applications/RevokeAccess.js [hunk 1]
```diff
@@ -13,7 +13,9 @@ import ConfirmLayout from '../../components/ConfirmLayout';
 
 const RevokeApplicationAccess = () => (
   <ConfirmLayout
-    question={l('Are you sure you want to revoke this application\'s access?')}
+    question={l(
+      'Are you sure you want to revoke this application\'s access?',
```
Comment (yvanzo) [inline root/account/applications/RevokeAccess.js]:
```suggestion
      `Are you sure you want to revoke this application's access?`,
```

File: root/main/404.js [hunk 2]
```diff
@@ -21,24 +21,32 @@ const _404 = ({$c, ...props}) => (
     <div id="content">
       <h1>{l('Page Not Found')}</h1>
       <p>
-        <strong>{l('Sorry, the page you\'re looking for does not exist.')}</strong>
+        <strong>
+          {l('Sorry, the page you\'re looking for does not exist.')}
```
Comment (yvanzo) [inline root/main/404.js]:
```suggestion
          {l(`Sorry, the page you're looking for does not exist.`)}
```

File: root/account/applications/Index.js [hunk 3]
```diff
@@ -72,7 +81,13 @@ const Index = ({applications, appsPager, tokens, tokensPager}: Props) => (
     <h2>{l('Authorized Applications')}</h2>
 
     <p>
-      {l('Some applications and websites support accessing private data from or submitting data to MusicBrainz but require your permission to access your account. These are the applications that you have authorized to access your MusicBrainz account. If you no longer use some of the applications, you can revoke their access.')}
+      {l(
+        `Some applications and websites support accessing private data from
+         or submitting data to MusicBrainz but require your permission to
+         access your account. These are the applications that you have
+         authorizedto access your MusicBrainz account. If you no longer use
```
Comment (yvanzo) [inline root/account/applications/Index.js]:
```suggestion
         authorized to access your MusicBrainz account. If you no longer use
```

File: root/layout/components/sidebar/ReleaseSidebar.js [hunk 4]
```diff
@@ -77,7 +82,8 @@ const ReleaseSidebar = ({$c, release}: Props) => {
             artwork={releaseArtwork}
             fallback={releaseCoverUrl}
             message={ReactDOMServer.renderToStaticMarkup(exp.l(
-              'Front cover image failed to load correctly.<br/>{all|View all artwork}.',
+              `Front cover image failed to load correctly.
+               <br/>{all|View all artwork}.`,
```
Comment (yvanzo) [inline root/layout/components/sidebar/ReleaseSidebar.js]: (For an unknown reason, the previous review is missing this comment.)

This is the only change that also changes localization message by adding a space before the `br` tag. Maybe it would be nice not to stop halfway, that is, to add a space after the `br` tag too:
```suggestion
               <br/> {all|View all artwork}.`,
```

File: root/static/scripts/common/components/CommonsImage.js [hunk 5]
```diff
@@ -52,4 +52,6 @@ class CommonsImage extends React.Component<Props, State> {
   }
 }
 
-export default hydrate<Props>('div.commons-image', CommonsImage, minimalEntity);
+export default hydrate<Props>(
+  'div.commons-image', CommonsImage, minimalEntity,
```
Comment (mwiencek) [inline root/static/scripts/common/components/CommonsImage.js]: When we have to move arguments to the next line (so no argument starts on the same line as the opening `(`), I'd actually prefer putting them all on separate lines. It'd seem easier to read to me that way, because if I see an initial line break, then I instinctively look for an argument per line and don't expect them to share a line.

If we enabled https://eslint.org/docs/rules/function-call-argument-newline and set it to "consistent," I think the style used here would even be an error, since the first argument has a line break before it while the others don't. (But this common style would still be allowed:

```JS
foo('bar', {
  baz: true,
});
```
since none of the arguments start with line breaks there.)

(This isn't the only instance of this change, I brought it up because there are many.)

File: root/layout/components/sidebar/ReleaseSidebar.js (hunk 4 above)
Comment (mwiencek) [reply root/layout/components/sidebar/ReleaseSidebar.js]: I would maybe use string concatenation here so we don't have to break the string to begin with

File: .eslintrc.yaml [hunk 6]
```diff
@@ -124,8 +124,8 @@ rules:
   lines-between-class-members: [warn, always]
   max-len: [warn, {code: 78,
                    ignoreUrls: true,
-                   ignoreStrings: true,
-                   ignoreTemplateLiterals: true,
+                   ignoreStrings: false,
+                   ignoreTemplateLiterals: false,
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply .eslintrc.yaml]: They probably could, I think they're (at least currently) false by default. I just thought specifying it made it more clear that it's intentional :)

File: root/account/applications/Index.js (hunk 3 above)
Comment (reosarevok) [reply root/account/applications/Index.js]: Good catch, sorry! :)

File: root/server/components.js [hunk 7]
```diff
@@ -14,7 +14,8 @@ module.exports = {
    */
   'account/Donation': require('../account/Donation'),
   'account/EditProfile': require('../account/EditProfile'),
-  'account/EmailVerificationStatus': require('../account/EmailVerificationStatus'),
+  'account/EmailVerificationStatus':
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply root/server/components.js]: I considered it and then thought this was hard to follow as-is, but if you do sort automatically, then I guess we can :)

File: root/account/applications/Index.js [hunk 8]
```diff
@@ -44,7 +50,10 @@ const buildTokenRow = (token: EditorOAuthTokenT, index: number) => (
     <td>{token.application.name}</td>
     <td>{formatScopes(token)}</td>
     <td>
-      <a href={'/account/applications/revoke-access/' + token.application.id + '/' + token.scope}>
+      <a
+        href={'/account/applications/revoke-access/' +
+          token.application.id + '/' + token.scope}
```
[the remark this answers is not in the document]
Comment (mwiencek) [reply root/account/applications/Index.js]: Decided this is probably fine (either style works for me, and at least you made them all consistent)

File: root/edit/components/Vote.js [hunk 9]
```diff
@@ -32,7 +32,9 @@ type VoteCheckboxProps = {
   +value: number,
 };
 
-const VoteCheckbox = ({edit, user, label, name, ...props}: VoteCheckboxProps) => {
+const VoteCheckbox = ({
+  edit, user, label, name, ...props
```
Comment (mwiencek) [inline root/edit/components/Vote.js]:
```suggestion
  edit,
  user,
  label,
  name,
  ...props
```

File: root/layout/index.js [hunk 10]
```diff
@@ -91,9 +102,12 @@ const Layout = ({$c, ...props}) => (
     <body>
       <Header {...props} />
 
-      {!getRequestCookie($c.req, 'server_details_dismissed_mtime') && <ServerDetailsBanner />}
+      {!getRequestCookie($c.req, 'server_details_dismissed_mtime') &&
+        <ServerDetailsBanner />}
 
-      {!!($c.stash.alert && $c.stash.alert_mtime > getRequestCookie($c.req, 'alert_dismissed_mtime', 0)) &&
+      {!!($c.stash.alert && $c.stash.alert_mtime > getRequestCookie(
+        $c.req, 'alert_dismissed_mtime', 0,
```
Comment (mwiencek) [inline root/layout/index.js]: Maybe break after the `>` above and do
```suggestion
          getRequestCookie($c.req, 'alert_dismissed_mtime', 0)) &&
```

File: root/layout/index.js [hunk 11]
```diff
@@ -119,32 +136,49 @@ const Layout = ({$c, ...props}) => (
         </div>}
 
       {!!($c.stash.new_edit_notes &&
-          $c.stash.new_edit_notes_mtime > getRequestCookie($c.req, 'new_edit_notes_dismissed_mtime', 0) &&
-          ($c.user.is_limited || getRequestCookie($c.req, 'alert_new_edit_notes', 'true') !== 'false')) &&
+          $c.stash.new_edit_notes_mtime > getRequestCookie(
+            $c.req, 'new_edit_notes_dismissed_mtime', 0,
```
Comment (mwiencek) [inline root/layout/index.js]: Same as above

File: root/layout/index.js [hunk 12]
```diff
@@ -119,32 +136,49 @@ const Layout = ({$c, ...props}) => (
...
+          ) &&
+          ($c.user.is_limited || getRequestCookie(
+            $c.req, 'alert_new_edit_notes', 'true',
```
Comment (mwiencek) [inline root/layout/index.js]: Same as above

File: root/release_group/ReleaseGroupIndex.js [hunk 13]
```diff
@@ -131,8 +131,11 @@ const ReleaseGroupIndex = ({
               <thead>
                 <tr>
                   {$c.user_exists
-                    ? <th className="checkbox-cell"><input type="checkbox" /></th>
-                    : null}
+                    ? (
```
Comment (mwiencek) [inline root/release_group/ReleaseGroupIndex.js]: The `? (` here can go on the previous line and the innards de-intended to match the `) :` below. Or `: null}` can be moved to its own line to match the `? <th`.

File: root/search/components/DocResults.js [hunk 14]
```diff
@@ -21,7 +21,9 @@ const DocResults = () => (
         src={'https://cse.google.com/cse.js?cx=' + encodeURIComponent(GOOGLE_CUSTOM_SEARCH)}
         type="text/javascript"
       />
-      {React.createElement('gcse:search', {enablehistory: 'true', queryparametername: 'query'})}
+      {React.createElement(
+        'gcse:search', {enablehistory: 'true', queryparametername: 'query'},
```
Comment (mwiencek) [inline root/search/components/DocResults.js]:
```suggestion
        'gcse:search',
        {enablehistory: 'true', queryparametername: 'query'},
```

File: root/static/scripts/common/components/Autocomplete.js [hunk 15]
```diff
@@ -66,7 +67,9 @@ class Autocomplete extends React.Component {
     }
 
     if (nextProps.hasOwnProperty('isLookupPerformed')) {
-      autocomplete.element.toggleClass('lookup-performed', !!nextProps.isLookupPerformed);
+      autocomplete.element.toggleClass(
+        'lookup-performed', !!nextProps.isLookupPerformed,
```
Comment (mwiencek) [inline root/static/scripts/common/components/Autocomplete.js]:
```suggestion
        'lookup-performed',
        !!nextProps.isLookupPerformed,
```

File: root/static/scripts/common/components/Autocomplete2/formatters.js [hunk 16]
```diff
@@ -159,7 +159,9 @@ function formatEvent(event: AutocompleteEventT) {
         (event.time || ''),
       ) : null}
 
-      {showRelatedEntities(l('Performers'), event.related_entities.performers)}
+      {showRelatedEntities(
+        l('Performers'), event.related_entities.performers,
```
Comment (mwiencek) [inline root/static/scripts/common/components/Autocomplete2/formatters.js]:
```suggestion
        l('Performers'),
        event.related_entities.performers,
```

File: root/static/scripts/common/components/TagEditor.js [hunk 17]
```diff
@@ -450,158 +452,181 @@ class TagEditor extends React.Component<TagEditorProps, TagEditorState> {
...
+            <p>{l('Nobody has tagged this yet.')}</p>
+          )}
+
+          {(positiveTagsOnly && !tags.every(isAlwaysVisible)) ? (
+            <>
+              {this.props.$c.user_exists ? (
+                <p>
+                  {l(
+                    `Tags with a score of zero or below,
+                    and tags that you’ve downvoted are hidden.`,
```
Comment (mwiencek) [inline root/static/scripts/common/components/TagEditor.js]:
```suggestion
                     and tags that you’ve downvoted are hidden.`,
```

File: root/static/scripts/common/components/TagEditor.js [hunk 18]
```diff
@@ -450,158 +452,181 @@ class TagEditor extends React.Component<TagEditorProps, TagEditorState> {
...
-                <a href="#" onClick={this.hideNegativeTags.bind(this)}>{l('Hide tags with a score of zero or below, and tags that you’ve downvoted.')}</a>
+                {l('All tags are being shown.')}
               </p>
-            ) : (
+              {this.props.$c.user_exists ? (
+                <p>
+                  <a href="#" onClick={this.hideNegativeTags.bind(this)}>
+                    {l(
+                      `Hide tags with a score of zero or below,
+                      and tags that you’ve downvoted.`,
```
Comment (mwiencek) [inline root/static/scripts/common/components/TagEditor.js]:
```suggestion
                       and tags that you’ve downvoted.`,
```

File: root/static/scripts/common/components/TagEditor.js [hunk 19]
```diff
@@ -450,158 +452,181 @@ class TagEditor extends React.Component<TagEditorProps, TagEditorState> {
...
+          ) : null}
+
+          {this.props.$c.user_exists ? (
+            <>
+              <h2>{l('Add Tags')}</h2>
               <p>
-                <a href="#" onClick={this.hideNegativeTags.bind(this)}>{l('Hide tags with a score of zero or below.')}</a>
+                {exp.l(
+                  `You can add your own {tagdocs|tags} below.
+                  Use commas to separate multiple tags.`,
```
Comment (mwiencek) [inline root/static/scripts/common/components/TagEditor.js]:
```suggestion
                   Use commas to separate multiple tags.`,
```

File: root/static/scripts/common/ratings.js [hunk 20]
```diff
@@ -46,11 +46,15 @@ $(document).on("click", "span.star-rating a", function () {
...
-            var newRating = data.rating == originalRating ? 0 : originalRating;
+            var newRating = data.rating == originalRating
+                ? 0
+                : originalRating;
             var oldRatingMatch = this.href.match(/rating=(\d+)/);
             if (oldRatingMatch[1] != newRating)
             {
-                this.href = this.href.replace(oldRatingMatch[0], 'rating=' + newRating);
+                this.href = this.href.replace(
+                    oldRatingMatch[0], 'rating=' + newRating,
```
Comment (mwiencek) [inline root/static/scripts/common/ratings.js]:
```suggestion
                    oldRatingMatch[0],
                    'rating=' + newRating,
```

File: root/static/scripts/common/utility/formatEndDate.js [hunk 21]
```diff
@@ -9,7 +9,10 @@
 
 import formatDate from './formatDate';
 
-export default function formatEndDate<+T: {...DatePeriodRoleT, ...}>(entity: T) {
+export default function formatEndDate<+T: {
+  ...DatePeriodRoleT,
+  ...
```
Comment (mwiencek) [inline root/static/scripts/common/utility/formatEndDate.js]:
```suggestion
  ...,
```

File: root/static/scripts/edit/ExampleRelationships.js [hunk 22]
```diff
@@ -119,16 +119,21 @@ RelationshipSearcher = function () {
         .done(function (data, status, jqxhr) {
             var search_result_type = data.entityType.replace("-", "_");
 
-            if (!(search_result_type === type0 || search_result_type === type1)) {
-                self.error('Invalid type for this relationship: ' +  search_result_type +
+            if (!(search_result_type === type0 ||
+                search_result_type === type1)) {
```
Comment (mwiencek) [inline root/static/scripts/edit/ExampleRelationships.js]:
```suggestion
                  search_result_type === type1)) {
```

File: root/static/scripts/edit/ExampleRelationships.js [hunk 23]
```diff
@@ -119,16 +119,21 @@ RelationshipSearcher = function () {
...
+                self.error('Invalid type for this relationship: ' +
+                            search_result_type +
```
Comment (mwiencek) [inline root/static/scripts/edit/ExampleRelationships.js]:
```suggestion
                           search_result_type +
```

File: root/static/scripts/edit/utility/linkPhrase.js [hunk 24]
```diff
@@ -100,7 +101,9 @@ export type LinkPhraseI18n<T> = {
 };
 
 const reactI18n: LinkPhraseI18n<Expand2ReactOutput> = {
-  cache: new WeakMap<RelationshipInfoT, CachedLinkPhraseData<Expand2ReactOutput>>(),
+  cache: new WeakMap<
+    RelationshipInfoT, CachedLinkPhraseData<Expand2ReactOutput>
```
Comment (mwiencek) [inline root/static/scripts/edit/utility/linkPhrase.js]:
```suggestion
    RelationshipInfoT,
    CachedLinkPhraseData<Expand2ReactOutput>,
```

File: root/statistics/stats.js [hunk 25]
```diff
@@ -744,28 +750,39 @@ for (let n = 0; n < 11; n++) {
     category: 'other',
     color: '#ff0000',
     description: texp.l('Count of all Releases with {n} Disc IDs', no),
-    label: texp.ln('Releases with 1 Disc ID', 'Releases with {n} Disc IDs', n, no),
+    label: texp.ln(
+      'Releases with 1 Disc ID', 'Releases with {n} Disc IDs', n, no,
```
Comment (mwiencek) [inline root/statistics/stats.js]:
```suggestion
      'Releases with 1 Disc ID',
      'Releases with {n} Disc IDs',
      n,
      no,
```

File: root/statistics/stats.js [hunk 26]
```diff
@@ -744,28 +750,39 @@ for (let n = 0; n < 11; n++) {
...
+    ),
   };
 
   stats[`count.medium.${n}discids`] = {
     category: 'other',
     color: '#ff0000',
     description: texp.l('Count of all Mediums with {n} Disc IDs', no),
-    label: texp.ln('Mediums with 1 Disc ID', 'Mediums with {n} Disc IDs', n, no),
+    label: texp.ln(
+      'Mediums with 1 Disc ID', 'Mediums with {n} Disc IDs', n, no,
```
Comment (mwiencek) [inline root/statistics/stats.js]:
```suggestion
      'Mediums with 1 Disc ID',
      'Mediums with {n} Disc IDs',
      n,
      no,
```

File: root/statistics/stats.js [hunk 27]
```diff
@@ -744,28 +750,39 @@ for (let n = 0; n < 11; n++) {
...
+    ),
   };
 
   stats[`count.recording.${n}releases`] = {
     category: 'other',
     color: '#ff0000',
     description: texp.l('Count of all Recordings with {n} Releases', no),
-    label: texp.ln('Recordings with 1 Release', 'Recordings with {n} Releases', n, no),
+    label: texp.ln(
+      'Recordings with 1 Release', 'Recordings with {n} Releases', n, no,
```
Comment (mwiencek) [inline root/statistics/stats.js]:
```suggestion
      'Recordings with 1 Release',
      'Recordings with {n} Releases',
      n,
      no,
```

File: root/tag/EntityList.js [hunk 28]
```diff
@@ -24,7 +25,9 @@ const headingsText = {
   place: N_ln('{num} place found', '{num} places found'),
   recording: N_ln('{num} recording found', '{num} recordings found'),
   release: N_ln('{num} release found', '{num} releases found'),
-  release_group: N_ln('{num} release group found', '{num} release groups found'),
+  release_group: N_ln(
+    '{num} release group found', '{num} release groups found',
```
Comment (mwiencek) [inline root/tag/EntityList.js]:
```suggestion
    '{num} release group found',
    '{num} release groups found',
```

File: root/tag/TagIndex.js [hunk 29]
```diff
@@ -75,17 +78,39 @@ const TagIndex = (props: Props) => (
...
-    {buildSection(props, 'instrument', l('Instruments'), N_ln('See all {num} instruments', 'See all {num} instruments'))}
-    {buildSection(props, 'label', l('Labels'), N_ln('See all {num} labels', 'See all {num} labels'))}
-    {buildSection(props, 'place', l('Places'), N_ln('See all {num} places', 'See all {num} places'))}
-    {buildSection(props, 'release_group', l('Release Groups'), N_ln('See all {num} release groups', 'See all {num} release groups'))}
-    {buildSection(props, 'release', l('Releases'), N_ln('See all {num} releases', 'See all {num} releases'))}
-    {buildSection(props, 'recording', l('Recordings'), N_ln('See all {num} recordings', 'See all {num} recordings'))}
-    {buildSection(props, 'series', l('Series'), N_ln('See all {num} series', 'See all {num} series'))}
-    {buildSection(props, 'work', l('Works'), N_ln('See all {num} works', 'See all {num} works'))}
+    {buildSection(props, 'area', l('Areas'), N_ln(
+      'See all {num} areas', 'See all {num} areas',
```
Comment (mwiencek) [inline root/tag/TagIndex.js]: (This suggestion applies to all `buildSection` calls in this file.)
```suggestion
      'See all {num} areas',
      'See all {num} areas',
```

File: root/utility/age.js [hunk 30]
```diff
@@ -153,7 +153,9 @@ export function displayAgeAgo(age: [number, number, number]) {
   if (years) {
     return texp.ln('{num} year ago', '{num} years ago', years, {num: years});
   } else if (months) {
-    return texp.ln('{num} month ago', '{num} months ago', months, {num: months});
+    return texp.ln(
+      '{num} month ago', '{num} months ago', months, {num: months},
```
Comment (mwiencek) [inline root/utility/age.js]:
```suggestion
      '{num} month ago',
      '{num} months ago',
      months,
      {num: months},
```

File: root/layout/index.js (hunk 12 above)
Comment (reosarevok) [reply root/layout/index.js]: Needs to be 

```
          ($c.user.is_limited ||
          getRequestCookie($c.req, 'alert_new_edit_notes', 'true') !==
          'false')) &&
```

to fit the lines, but guess that'd still be better.

**Comment by mwiencek** [comment]
Took another look and made some suggested improvements. :) See the force-push.

The changes are primarily in the added commits, though I amended your max-len one to change back the one line in root/account/applications/Index.js since we kept all the others to use the previous style, and to remove some trailing whitespace in the `CDStubSidebar` `searchQuery`.

The commit for root/statistics/Index.js is the main thing though - I wasn't too happy with how that file exploded so I added some shortcuts functions as an alternative (which ended up removing 550 lines from this PR).

File: root/layout/components/sidebar/CDStubSidebar.js [hunk 31]
```diff
@@ -35,11 +36,19 @@ const CDStubSidebar = ({cdstub}: Props) => {
...
+    ? escapeLuceneValue(cdstub.barcode)
+    : null;
+
   const searchQuery = (
-    'artist:(' + escapeLuceneValue(cdstub.artist || l('Various Artists')) + ') ' +
-    'release:(' + escapeLuceneValue(cdstub.title) + ') ' +
-    'tracksmedium:(' + escapeLuceneValue(cdstub.track_count) + ')' +
-    (cdstub.barcode ? ' barcode:(' + escapeLuceneValue(cdstub.barcode) + ')' : '')
+    `artist:(${artistField}) ` +
+    `release:(${releaseField} ` +
```
Comment (yvanzo) [inline root/layout/components/sidebar/CDStubSidebar.js]: Closing parenthesis went missing since commit https://github.com/metabrainz/musicbrainz-server/pull/1248/commits/2170478f2d73555571f7140acb0e5b34301f7e74
```suggestion
    `release:(${releaseField}) ` +
```

Comment (mwiencek) [reply root/layout/components/sidebar/CDStubSidebar.js]: Good catch, fixed

File: root/statistics/Index.js [hunk 32]
```diff
@@ -72,69 +89,69 @@ const Index = ({
...
           <tr>
             <th>{l('Artists:')}</th>
-            <td colSpan="3">{formatCount($c, stats['count.artist'])}</td>
+            <td colSpan="3">{fc('artist')}</td>
           </tr>
           <tr>
             <th>{l('Release Groups:')}</th>
             <td colSpan="3">
-              {formatCount($c, stats['count.releasegroup'])}
+              {fc('releasegroup')}
```
Comment (reosarevok) [inline root/statistics/Index.js]: Maybe we should now make it so that all the {fc()} calls are in one line with the tds, unless they don't fit at all? This looks like it'd fit.

Comment (mwiencek) [reply root/statistics/Index.js]: Yep, that'd make sense. Fixed all the ones I saw.
